### PR TITLE
add `collected.count` and `collected.max`

### PR DESCRIPTION
**Pull request by WanderingFox**
Adds a `collected.count` and a `collected.max` condition that return the amount of pets captured and the max available to capture respectively.

This adds more of the functionality that was asked for on tdbattlescripts to manage capturing. It lets you figure out how many pets you have captured and then go from there. I'm not entirely sure why someone would want `collected.max` since this is only really useful for wild pet capture, but I included it for completeness.

Example:

```
if [enemy.collected.count < 3]
   ...
endif
```

I'm still hoping to figure out how to implement `self.collected.quality.min` or similar, but it's currently looking like that will require scanning the entire pet journal, which I'm hoping to avoid.

### CODE REVIEW

**Comment by axc450** [comment]
Can you please add a changelog entry for this?

**Comment by axc450** [comment]
@bloerwald Do you have any comments on the non-existability of these selectors (ie https://github.com/DengSir/tdBattlePetScript/issues/26)? As this is a global selector, I think this selector will always exist so is probably not a concern.

**Comment by axc450** [comment]
@WanderingFox API docs would also be nice for this! 😄

**Comment by bloerwald** [comment]
> @bloerwald Do you have any comments on the non-existability of these selectors (ie [DengSir/tdBattlePetScript#26](https://github.com/DengSir/tdBattlePetScript/issues/26))? As this is a global selector, I think this selector will always exist so is probably not a concern.

Yes, it will probably break if the pet can't be found, i.e. if `select(2, C_PetJournal.FindPetIDByName(name)) = nil`.

File: Extension/Conditions.lua [hunk 1]
```diff
@@ -202,6 +202,18 @@ Addon:RegisterCondition('collected', { type = 'boolean', arg = false }, function
...
 
+Addon:RegisterCondition('collected.count', { type = 'compare', arg = false }, function(owner, pet)
+    local species = C_PetJournal.FindPetIDByName(select(2, C_PetBattles.GetName(owner, pet)))
+    local collected = C_PetJournal.GetNumCollectedInfo(species)
+    return collected
+end)
+
+Addon:RegisterCondition('collected.max', { type = 'compare', arg = false }, function(owner, pet)
+    local species = C_PetJournal.FindPetIDByName(select(2, C_PetBattles.GetName(owner, pet)))
+    local max = select(2, C_PetJournal.GetNumCollectedInfo(species))
```
Comment (bloerwald) [inline Extension/Conditions.lua:213]:
```suggestion
    local max = species and select(2, C_PetJournal.GetNumCollectedInfo(species)) or 0
```

File: Extension/Conditions.lua [hunk 2]
```diff
@@ -202,6 +202,18 @@ Addon:RegisterCondition('collected', { type = 'boolean', arg = false }, function
     return select(2, C_PetJournal.FindPetIDByName(name)) ~= nil
 end)
 
+Addon:RegisterCondition('collected.count', { type = 'compare', arg = false }, function(owner, pet)
+    local species = C_PetJournal.FindPetIDByName(select(2, C_PetBattles.GetName(owner, pet)))
+    local collected = C_PetJournal.GetNumCollectedInfo(species)
```
Comment (bloerwald) [inline Extension/Conditions.lua:207]:
```suggestion
    local collected = species and select(1, C_PetJournal.GetNumCollectedInfo(species)) or 0
```

**Comment by axc450** [comment]
Also have a think about how these should act against pets that you cant actually obtain (ie elite pets, trainer pets - basically anything that isn't wild). I would like `collected.max` to return `3` for wild pets, `1` for unique pets and `0` for unobtainable pets.

**Comment by WanderingFox** [comment]
Made the suggested changes to catch nil pets, and added an obtainable check which forces the values to 0 if the pet is unobtainable.

@axc450 the docs are in the open issue ( #3 )

**Comment by axc450** [comment]
Looks great! Thanks!
@bloerwald Any further comments?

**Comment by bloerwald** [comment]
Looks fine to me, but I only looked at the code. Ship it!

**Review by bloerwald — APPROVED** [review]

**Pull request merged**